### PR TITLE
Puppetfile: update all the versions

### PR DESCRIPTION
**Pull request by daenney**

### CODE REVIEW

**Comment by daenney** [comment]
Supersedes #921

File: Puppetfile [hunk 1]
```diff
@@ -7,25 +7,23 @@ moduledir File.join(File.dirname(__FILE__), 'modules')
...
-mod 'puppetlabs-host_core', '1.0.1'
-mod 'puppetlabs-scheduled_task', '1.0.0'
-mod 'puppetlabs-sshkeys_core', '1.0.1'
-mod 'puppetlabs-zfs_core', '1.0.1'
-mod 'puppetlabs-cron_core', '1.0.0'
-mod 'puppetlabs-mount_core', '1.0.2'
-mod 'puppetlabs-selinux_core', '1.0.1'
-mod 'puppetlabs-yumrepo_core', '1.0.1'
-mod 'puppetlabs-zone_core', '1.0.1'
+mod 'puppetlabs-augeas_core', '1.0.4'
```
Comment (donoghuc) [inline Puppetfile]: Updating the puppet core modules makes sense to me.

File: Puppetfile [hunk 2]
```diff
@@ -7,25 +7,23 @@ moduledir File.join(File.dirname(__FILE__), 'modules')
 # Core modules used by 'apply'
 mod 'puppetlabs-service', '0.6.0'
 mod 'puppetlabs-facts', '0.5.1'
-mod 'puppet_agent',
```
Comment (donoghuc) [inline Puppetfile]: I have not had a chance to verify the new `puppet_agent::install` task parameters https://github.com/puppetlabs/puppetlabs-puppet_agent/commit/358add8dfaff091f4f9346e47652fbdb764cb909#diff-11a53f781e47039ffcac93b25abd1e81

Comment (donoghuc) [reply Puppetfile]: I think shipping these is low risk. Acceptance tests for 2.1.1 are green and while there are not yet tests for the new parameters they are unlikely to cause regressions.

**Pull request merged**